### PR TITLE
Optimize ArbitraryMotion

### PR DESCRIPTION
**Pull request by pvillacorta**
This PR is intended to address #371 and #372

### CODE REVIEW

**Comment by pvillacorta** [comment]
I created a new branch [optimize-arbitrary-parallel](https://github.com/JuliaHealth/KomaMRI.jl/tree/optimize-arbitrary-parallel) in which I initialize the interpolation functions and store them in the `ExplicitArbitraryMotion` struct. This does not mean pre calculating the trajectories, it just means initializing them, but the interpolations are still done "on the fly". This way, we avoid a lot of CPU->GPU transfers. The results are even better than before:
` 0.160241 seconds (86.02 k allocations: 2.652 MiB)`

**Comment by pvillacorta** [comment]
A new branch, [optimize-arbitrary-view](https://github.com/JuliaHealth/KomaMRI.jl/tree/optimize-arbitrary-view), was created. This branch initializes the interpolating functions directly in GPU, so CPU-GPU transfers are not needed for this now. The performance is even better, although with a bit more allocations than [optimize-arbitrary-parallel](https://github.com/JuliaHealth/KomaMRI.jl/tree/optimize-arbitrary-parallel):

` 0.135876 seconds (95.69 k allocations: 2.986 MiB)`

In addition, a custom adapt function for ArbitraryMotion is not needed now. Maybe this can solve problems in #405 and #406. My branch has not included the changes from these last two PRs from @rkierulf  yet, so maybe I will need to change something once this is done.

In short, since [optimize-arbitrary-view](https://github.com/JuliaHealth/KomaMRI.jl/tree/optimize-arbitrary-view) is by far the preferred option, I will close this PR and open a new one with this branch.
